packages/react/src/analytics/integrations/GA4/GA4.ts

```diff
@@ -148,9 +150,36 @@ class GA4 extends integrations.Integration<GA4IntegrationOptions> {
       true,
     );
 
+    const dataLayer = get(
```

Comment:
Since this is being used in `setConsent` as well, what do you think of creating a helper function called something like `resolveDataLayerName` which receives the options as an argument and returns the datalayer name which will be used?